server/notifications_slack.go

```diff
@@ -314,3 +314,91 @@ func hideSlackAuthToken(authToken string) string {
 	}
 	return authToken
 }
+
+type slackMsg struct {
+	Msg          []byte
+	ReplySubject string
+}
+
+func sendSlackNotifications(s *Server, msgs []slackMsg) {
+	tenantMsgs := groupMessagesByTenant(msgs, s)
+	for tenantName, tMsgs := range tenantMsgs {
+		sendTenantSlackNotifications(s, tenantName, tMsgs)
+	}
+}
+
+func sendTenantSlackNotifications(s *Server, tenantName string, msgs []NotificationMsgWithReply) {
+	var ok bool
+	if _, ok := NotificationFunctionsMap[slackIntegrationName]; !ok {
+		s.Errorf("slack integration doesn't exist")
+		return
+	}
+
+	var tenantIntegrations map[string]any
+	if tenantIntegrations, ok = IntegrationsConcurrentCache.Load(tenantName); !ok {
+		// slack is either not enabled or have been disabled - just ack these messages
+		ackMsgs(s, msgs)
+		return
+	}
+
+	var slackIntegration models.SlackIntegration
+	if slackIntegration, ok = tenantIntegrations[slackIntegrationName].(models.SlackIntegration); !ok {
+		// slack is either not enabled or have been disabled - just ack these messages
+		ackMsgs(s, msgs)
+		return
+	}
+
+	for i := 0; i < len(msgs); i++ {
+		m := msgs[i]
+		err := sendMessageToSlackChannel(slackIntegration, m.NotificationMsg.Title, m.NotificationMsg.Message)
+		if err != nil {
+			s.Errorf("failed to send slack notification: " + err.Error())
```

Comment:
in case the channel was not found we should log this as a warn rather than an error